src/state/thunkMiddleware.ts

```diff
@@ -48,3 +52,17 @@ export const requestResource = (resource: String.AsObject) => {
     }
   };
 };
+
+// FETCH DATA TYPES
+export const requestDataTypes = (codeSystemName: string = "data-types") => {
+  return async (dispatch: ThunkDispatch<RootState, void, Action>) => {
+    const response: AxiosResponse<any> = await api.get(
```

Comment:
If you want to update the 'loading' state attribute in your 'codeSystem' reducer, it would be from here by dispatching an action like 'getCodeSystemDataTypePending'